### DISCUSSION TITLE
Storyshots and Controls

### DISCUSSION THREAD

**Discussion by fictivecreations** [storybookjs/storybook#13865] [Help] [answered]
If my storybook has both the controls addon and the storyshot addon will a storyshot be generated for every permutation of the controls? Or will I have to make stories for that manually?

**Comment by shilman** [comment, marked as answer]
You need to make stories by hand. There's a proposal for combinatorial testing here, but as yet unimplemented

https://docs.google.com/document/d/1MF1QZdMvJ-0mMxCuFXxgthkig5AZp2L6f4VGNIu03zc/edit?usp=sharing

**Reply by jorisw** [reply]
I'm not sure that we're looking for Storybook Combos per se.

We're looking for StoryShots to cover boolean and enum Control options.

This would automate 10x the coverage on the components, without having a Combo story, which is a feature with a different target audience.
